src/funcade/middleware/buddy.clj

```diff
@@ -39,13 +40,22 @@
 
     (-authenticate [_ request data]
       (try
-        (let [tkey (-> data
-                       jwks/find-kid
-                       jwks/find-token-key-by-kid)]
-          (when-not tkey
-            (throw (ex-info "jwt token is signed by unknown key (i.e. no public key in JSON Web Key Sets to verify the signature)"
-                            {:type :validation :cause :incorrect-sign-key})))
-          (authfn (jwt/unsign data tkey options)))
+        (letfn [(validate-token ([token-data]
+                                 (-> token-data
+                                     jwks/find-kid
+                                     jwks/find-token-key-by-kid)))
+                (authenticate-request ([token-key retry?]
+                                       (if token-key
+                                         (authfn (jwt/unsign data token-key options))
+                                         (let [unauthorized-error (ex-info "jwt token is signed by unknown key (i.e. no public key in JSON Web Key Sets to verify the signature)"
+                                                                           {:type :validation :cause :incorrect-sign-key})]
+                                           (if retry? 
+                                             (do (jwks/refresh-kids uri nil) ;; nil is to avoid refresh scheduler
```

Comment:
looks like [refresh-kids](https://github.com/tolitius/funcade/blob/48a8f433e570d4ec4d97f622d91c62f3e73ae67d/src/funcade/jwks.clj#L54-L62) needs to be updated to something like:
```clojure
(defn refresh-kids
  "refresh jwks keyset from auth provider"
  ([uri]
   (refresh-kids uri nil))
  ([uri callback]
   (let [current-kids      (-> uri jwks->keys keys) 
         refresh-callback  (or callback
                               (fn [_]
                                 (prn "[funcade]: refreshed keyset")))] 
     (refresh-callback current-kids)
     current-kids)))
```

so we can just do:
```clojure
(jwks/refresh-kids uri)
```